cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/field_display_filtering.spec.js

```diff
@@ -0,0 +1,245 @@
+        dataExplorer
+          .getExpandedDocTableRow(docTableRowNumber, expandedDocumentRowNumber)
+          .within(() => {
+            cy.getElementByTestId('addExistsFilterButton').click();
+          });
+        // Verify full pill text
+        // globalFilterLabelValue gives the inner element, but we may want all the text in the filter pill
+        cy.getElementByTestId('globalFilterLabelValue', {
+          timeout: 10000,
+        })
+          .parent()
+          .should('have.text', filterFieldText + ': ' + 'exists');
+        cy.getElementByTestId('discoverQueryHits').should(
+          'have.text',
+          expectedQueryHitsAfterFilterApplied
+        );
+      });
+    cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+    cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+  };
+
+  selectDataset(datasetType, language);
+  setDateRange(datasetType, language);
+
+  cy.getElementByTestId('docTable').get('tbody tr').should('have.length.above', 3); // To ensure it waits until a full table is loaded into the DOM, instead of a bug where table only has 1 hit.
+  dataExplorer.toggleDocTableRow(0);
+  verifyFirstExpandedFieldFilterForFilterOutFilterExistsButtons();
+  dataExplorer.verifyDocTableFirstExpandedFieldFirstRowToggleColumnButtonHasIntendedBehavior();
+
+  if (isFilterButtonsEnabled) {
+    verifyDocTableFirstExpandedFieldFirstRowFilterForOutButtonFiltersCorrectField(
+      'for',
+      0,
+      0,
+      '10,000',
+      '1'
+    );
+    verifyDocTableFirstExpandedFieldFirstRowFilterForOutButtonFiltersCorrectField(
+      'out',
+      0,
+      0,
+      '10,000',
+      '9,999'
+    );
+    verifyDocTableFirstExpandedFieldFirstRowExistsFilterButtonFiltersCorrectField(
+      0,
+      0,
+      '10,000',
+      '10,000'
+    );
+  }
+};
+describe('filter for value spec', () => {
+  before(() => {
+    // Load test data
+    cy.setupTestData(
+      SECONDARY_ENGINE.url,
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.mapping.json'],
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.data.ndjson']
+    );
+
+    // Add data source
+    cy.addDataSource({
+      name: `${DATASOURCE_NAME}`,
+      url: `${SECONDARY_ENGINE.url}`,
+      authType: 'no_auth',
+    });
+    // Create workspace
+    cy.deleteWorkspaceByName(`${workspace}`);
+    cy.visit('/app/home');
+    cy.createInitialWorkspaceWithDataSource(`${DATASOURCE_NAME}`, `${workspace}`);
+    cy.wait(2000);
+    cy.createWorkspaceIndexPatterns({
+      url: `${BASE_PATH}`,
+      workspaceName: `${workspace}`,
+      indexPattern: INDEX_NAME,
+      timefieldName: 'timestamp',
+      indexPatternHasTimefield: true,
+      dataSource: DATASOURCE_NAME,
+      isEnhancement: true,
+    });
+  });
+
+  beforeEach(() => {
+    cy.navigateToWorkSpaceHomePage(`${BASE_PATH}`, `${workspace}`);
```

Comment:
can we change this to `navigateToWorkSpaceSpecificPage` with `page: 'discover'`? See https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9112#discussion_r1899810614 for rationale